cristin-project/src/main/java/no/unit/nva/cristin/projects/FetchOneCristinProject.java

```diff
@@ -71,11 +71,11 @@ protected Integer getSuccessStatusCode(Void input, NvaProject output) {
         return HttpURLConnection.HTTP_OK;
     }
 
-    private String getValidId(RequestInfo requestInfo) throws BadRequestException {
-        attempt(() -> Integer.parseInt(requestInfo.getPathParameter(ID)))
-                .orElseThrow(failure -> new BadRequestException(ERROR_MESSAGE_INVALID_PATH_PARAMETER_FOR_ID));
+    public static String getValidIdentifier(RequestInfo requestInfo) throws BadRequestException {
+        attempt(() -> Integer.parseInt(requestInfo.getPathParameter(IDENTIFIER)))
```

Comment:
I'm assuming that attempt catches both NPE and Number error, I would be tempted to return the value of the declarative code, but this involves a weird stringification at the end.